editor/plugins/resource_preloader_editor_plugin.cpp

```diff
@@ -229,17 +229,32 @@ void ResourcePreloaderEditor::_update_library() {
 		ERR_CONTINUE(r.is_null());
 
 		ti->set_tooltip(0, r->get_path());
-		String type = r->get_class();
-		ti->set_text(1, type);
+		ti->set_text(1, r->get_path());
+		ti->add_button(1, get_icon("InstanceOptions", "EditorIcons"), BUTTON_SUBSCENE, false, TTR("Open in Editor"));
+		ti->set_tooltip(1, TTR("Instance:") + " " + r->get_path() + "\n" + TTR("Type:") + " " + r->get_class());
+		ti->set_editable(1, false);
 		ti->set_selectable(1, false);
+		String type = r->get_class();
+		ti->set_text(2, type);
+		ti->set_selectable(2, false);
 
 		if (has_icon(type, "EditorIcons"))
-			ti->set_icon(1, get_icon(type, "EditorIcons"));
+			ti->set_icon(2, get_icon(type, "EditorIcons"));
 	}
 
 	//player->add_resource("default",resource);
 }
 
+void ResourcePreloaderEditor::_cell_button_pressed(Object *p_item, int p_column, int p_id) {
+
+	TreeItem *item = Object::cast_to<TreeItem>(p_item);
```

Comment:
If this is always a `TreeItem`, shouldn't the method signature be typed as `Ref<TreeItem>` instead?

In any case, I would an `ERR_FAIL_COND` to be on the safe side.

Reply:
> If this is always a `TreeItem`, shouldn't the method signature be typed as `Ref<TreeItem>` instead?
I am not certain to understand. I took the example from scene_tree_editor.cpp (https://github.com/godotengine/godot/blob/34c988cfa92f19c232b65990704816ba1c7d2622/editor/scene_tree_editor.cpp#L48) and made it similar.

Ok, I put the `ERR_FAIL_COND` back.

Reply:
Back to your remark.
I don't know how I can cast the input of `_cell_button_pressed()` *before* it is called. It is connected to a `button_pressed` signal (here: https://github.com/godotengine/godot/pull/15713/files#diff-1704455f99063b3e8f3e99d9b2364720R406) and I looked at many places in the whole project and every callback function takes the same parameters.

Reply:
Oh, so the `button_pressed` signature has an `Object*`. It's weird to me, but such change would be out of the scope of this PR, so it's okay the way it is.